brianlong/www.validators.app
# 185887385 update avatar logic

#### What's this PR do?
- Saves avatar URL from solana CLI if present
- Displays keybase avatar only if avatar is not present in solana CLI (in that case, user must provide keybase_id)
- If both attributes (avatar link and keybase id) are present in CLI, app displays the link from solana CLI and ingnores the keybase (we don't check keybase at all if avatar link is already present)
- Updates published_information_score - gives 0.5 point for avatar presence, instead of keybase_id presence

#### How should this be manually tested?
LOCALLY:
- Upgrade Solana on your local machine to 1.16.9 (https://github.com/brianlong/www.validators.app#solana-cli)
- Select a validator without cli image url (verify by running `solana validator-info get --url https://api.rpcpool.com/20dc90af2aed522cdff09bab309e | grep -A 8 DDnAqxJVFo2GVTujibHt5cjevHMSE9bo8HJaydHoshdp`)
- Save an image manually `Validator.find_by(account: "DDnAqxJVFo2GVTujibHt5cjevHMSE9bo8HJaydHoshdp").update(avatar_url: "https://s3.amazonaws.com/keybase_processed_uploads/3af995d21a8fe4cec4d6e83104f87205_360_360.jpg")`
- Run `rails r script/rails r script/validators_get_info.rb`
- Check the validator given above and confirm that the avatar link was not overrided with empty value (keybase avatar is used)
- Confirm there're no errors in the script, check the logs
- Confirm that validators with CLI image url now have some avatars assigned and correctly displayed on various pages, eg. B4dn3WWS95M4qNXaR5NTdkNzhzvTZVqC13E3eLrWhXLa (mainnet), TSTdidhZVFcTwbYC2DtTRsZnWgGvz2W7cAe2oBRHcFH (testnet). 
- Confirm they have correct published_information_score assigned, eg. B4dn3WWS95M4qNXaR5NTdkNzhzvTZVqC13E3eLrWhXLa doesn't have keybase_id, but it has avatar_url from CLI, so the scrore should be 2. 

ON STAGING:
- Solana version is already upgraded
- validators_get_info.rb script runs once an hour, 5 minutes after full hour, so you can run it manually if you don't want to wait: `RAILS_ENV=stage bundle exec rails r script/validators_get_info.rb`. Then wait a few minutes until the scores get updated
- Check the avatars same as on local environment

#### What are the relevant tickets?
- [URL to the PivotalTracker ticket](https://www.pivotaltracker.com/story/show/185887385)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?
